tests/test_session.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+import pyama.session.model as pysm
+import pyama.session.controller as pysc
+import pyama.session.view_tk as pysvtk
+import pyama.session.sessionopener_tk as pyssotk
+class TestSession:
+    @pytest.fixture
+    def Controller(self):
+        return pysc.SessionController()
+    @pytest.fixture
+    def View(self, Controller):
+        view = pysvtk.SessionView_Tk(Controller.title, Controller.control_queue, Controller.status)
+        Controller.view = view
+        return view
+    @pytest.fixture
+    def Session(self, View, Controller):
+        View._session_opener = pyssotk.SessionOpener(View.root, control_queue=View.control_queue, status=View.status)
+        Controller.initialize_session()
```

Comment:
@dwoschee do you know why this does not set the `session_id` of the stackopener?

Reply:
As far as I can see, the test doesn’t call the `SessionController.start` method, which starts the queue consuming loops.

During normal startup, this method is called in the [`main` method](https://github.com/SoftmatterLMU-RaedlerGroup/pyama/blob/c6b7efec9e52e30ad8412f9e381306b4faabfed8/src/pyama/main.py#L25).

Reply:
yeah, I didn't want to actually start the gui. But even when I call `Controller.control_loop()` before I still get `None` for the session_id in the SessionOpener while it is correctly captured in the `SessionModel`

Reply:
You also need to call `View.poll_event_queue()` after `Controller.initialize_session()`.

Reply:
:+1: that works. Thanks!